src/soot/jimple/spark/solver/OnFlyCallGraph.java

```diff
@@ -72,15 +77,15 @@ public void build() {
     private void processReachables() {
         reachableMethods.update();
         while(reachablesReader.hasNext()) {
-            MethodOrMethodContext m = (MethodOrMethodContext) reachablesReader.next();
```

Comment:
This ocde is quite old, and back in the old days, there were no generics in Java. Consequently, we still have some pointless casts in the code that date back to times when everything was java.lang.Object.